src/librustc_trait_selection/traits/error_reporting/mod.rs

```diff
@@ -289,7 +289,7 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                             (
                                 Some(format!(
                                     "`?` couldn't convert the error to `{}`",
-                                    trait_ref.self_ty(),
+                                    trait_ref.skip_binder().self_ty(),
```

Comment:
"Less yay" -- this seems ok for now I guess but I wish we had a different solution than skipping the binder